src/Cameras/Inputs/BaseCameraPointersInput.ts

```diff
@@ -0,0 +1,328 @@
+                      multiTouchPanPosition);
+
+                    previousMultiTouchPanPosition = {x: (pointA.x + pointB.x) / 2,
+                                                     y: (pointA.y + pointB.y) / 2,
+                                                     pointerId: evt.pointerId,
+                                                     type: p.type};
+                    previousPinchSquaredDistance = pinchSquaredDistance;
+                }
+            }
+        };
+
+        this._observer = this.camera.getScene().onPointerObservable.add(
+            this._pointerInput,
+            PointerEventTypes.POINTERDOWN | PointerEventTypes.POINTERUP |
+            PointerEventTypes.POINTERMOVE);
+
+        this._onLostFocus = () => {
+            this.log("FollowCameraPointersInput._onLostFocus");
+            pointA = pointB = null;
+        };
+
+        element.addEventListener("contextmenu",
+            <EventListener>this.onContextMenu.bind(this), false);
+        element.addEventListener("MSPointerDown",
+            <EventListener>this.onGestureStart.bind(this), false);
+        element.addEventListener("MSGestureChange",
+            <EventListener>this.onGesture.bind(this), false);
+
+        Tools.RegisterTopRootEvents([
+            { name: "blur", handler: this._onLostFocus }
+        ]);
+    }
+
+    /**
+     * Detach the current controls from the specified dom element.
+     * @param element Defines the element to stop listening the inputs from
+     */
+    public detachControl(element: Nullable<HTMLElement>): void {
+        if (this._onLostFocus) {
+            Tools.UnregisterTopRootEvents([
+                { name: "blur", handler: this._onLostFocus }
+            ]);
+        }
+
+        if (element && this._observer) {
+            this.camera.getScene().onPointerObservable.remove(this._observer);
+            this._observer = null;
+
+            if (this.onContextMenu) {
+                element.removeEventListener("contextmenu", this.onContextMenu);
+            }
+
+            if (this.onGestureStart) {
+                element.removeEventListener("MSPointerDown", <EventListener>this.onGestureStart);
+            }
+
+            if (this.onGesture) {
+                element.removeEventListener("MSGestureChange", <EventListener>this.onGesture);
+            }
+
+            this._MSGestureHandler = null;
+            this._onLostFocus = null;
+        }
+    }
+
+    /**
+     * Gets the class name of the current input.
+     * @returns the class name
+     */
+    public getClassName(): string {
+        return "ArcRotateCameraPointersInput";
+    }
+
+    /**
+     * Get the friendly name associated with the input class.
+     * @returns the input friendly name
+     */
+    public getSimpleName(): string {
+        return "pointers";
+    }
+
+    protected doDoubleTouch(type: string) {
+        this.log("FollowCameraPointersBase.doDoubleTouch(", type, ")");
```

Comment:
we have to remove all logs for final review

Reply:
as above.